### PR TITLE
Generate SQL RM schemas

### PR DESCRIPTION
**Pull request by jaredmoo**

### CODE REVIEW

File: schemas/2014-04-01/servers.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+API Version: 2014-04-01
+## Template format
+
+To create a Microsoft.Sql/servers resource, add the following JSON to the resources section of your template.
+
+```json
+{
+  "name": "string",
+  "type": "Microsoft.Sql/servers",
+  "apiVersion": "2014-04-01",
```
Comment (TianoMS) [inline schemas/2014-04-01/servers.md]: >2014-04-01 [](start = 17, length = 10)

I don't see Microsoft.Sql/servers definied in 2014-04-01/Microsoft.Sql.json.

File: schemas/2014-04-01/servers_firewallRules.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Microsoft.Sql/servers/firewallRules template reference
+API Version: 2014-04-01
+## Template format
+
+To create a Microsoft.Sql/servers/firewallRules resource, add the following JSON to the resources section of your template.
+
+```json
+{
+  "name": "string",
+  "type": "Microsoft.Sql/servers/firewallRules",
```
Comment (TianoMS) [inline schemas/2014-04-01/servers_firewallRules.md]: I don't see firewallRules definition either.

**Comment by TianoMS** [comment]
Please add tests for the new schema.

File: tests/2014-04-01/Microsoft.Sql.tests.json [hunk 3]
```diff
@@ -0,0 +1,583 @@
...
+    {
+      "name": "servers - empty",
+      "definition": "http://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#/resourceDefinitions/servers",
+      "json":
+        {
+          "apiVersion": "2014-04-01",
+          "properties": { }
+        },
+        "expectedErrors":
+[
```
Comment (TianoMS) [inline tests/2014-04-01/Microsoft.Sql.tests.json]: please format this file.

File: schemas/2014-04-01/Microsoft.Sql.json [hunk 4]
```diff
@@ -0,0 +1,1388 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.Sql",
+  "description": "Microsoft Sql Resource Types",
+  "resourceDefinitions": {
+    "servers": {
```
Comment (TianoMS) [inline schemas/2014-04-01/Microsoft.Sql.json]: >servers [](start = 5, length = 7)

Please add all the top level resource types to schemas/2015-01-01/deploymentTemplate.json and schemas/2014-04-01-preview.json.

Comment (jaredmoo) [inline schemas/2014-04-01/Microsoft.Sql.json]: Done

File: schemas/2015-01-01/deploymentTemplate.json [hunk 5]
```diff
@@ -277,7 +277,26 @@
...
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#/resourceDefinitions/servers_databases_extensions" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#/resourceDefinitions/servers_databases_geoBackupPolicies" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#/resourceDefinitions/servers_databases_securityAlertPolicies" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#/resourceDefinitions/servers_elasticPools" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-04-01/Microsoft.Sql.json#/resourceDefinitions/servers_firewallRules" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-05-01-preview/Microsoft.Sql.json#/resourceDefinitions/servers" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2014-05-01-preview/Microsoft.Sql.json#/resourceDefinitions/servers_databases_auditingSettings" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >2014-05-01-preview [](start = 72, length = 18)

2015-05-01-preview?

Comment (jaredmoo) [inline schemas/2015-01-01/deploymentTemplate.json]: Yes, good catch

File: schemas/2014-04-01/servers.md (hunk 1 above)
Comment (jaredmoo) [inline schemas/2014-04-01/servers.md]: Fixed by generating properly.

File: schemas/2014-04-01/servers_firewallRules.md (hunk 2 above)
Comment (jaredmoo) [inline schemas/2014-04-01/servers_firewallRules.md]: Fixed by generating properly.

File: schemas/2015-05-01-preview/Microsoft.Sql.json [hunk 6]
```diff
@@ -0,0 +1,1185 @@
...
+                "type": "string"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Resource tags."
+        },
+        "identity": {
```
Comment (TianoMS) [inline schemas/2015-05-01-preview/Microsoft.Sql.json]: >identity [](start = 9, length = 8)

do we need this? I don't see it in any other resource type.

**Pull request merged**